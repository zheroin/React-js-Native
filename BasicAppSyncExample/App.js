import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'
import ListItems from './Components/listItems'
import AddItems from './Components/addItems'
// The AppSync Import Bus
import AWSAppSyncClient from 'aws-appsync'
import { Rehydrated } from 'aws-appsync-react'
import { AUTH_TYPE } from 'aws-appsync/lib/link/auth-link'
import { ApolloProvider } from 'react-apollo'
import AppSync from './AppSync.js'

const client = new AWSAppSyncClient({
  url: AppSync.graphqlEndpoint,
  region: AppSync.region,
  auth: {
    type: AUTH_TYPE.API_KEY,
    apiKey: AppSync.apiKey
  }
})

class App extends Component {
  state = { allItems: [] }
  render () {
    return (
      <ScrollView
        contentContainerStyle={styles.container}
        keyboardShouldPersistTaps='always'
      >
        <Text style={styles.welcome}>
          Welcome to Amazon AppSync
        </Text>
        <AddItems />
        <ListItems />
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})

const WithProvider = () => (
  <ApolloProvider client={client}>
    <Rehydrated>
      <App />
    </Rehydrated>
  </ApolloProvider>
)

export default WithProvider
