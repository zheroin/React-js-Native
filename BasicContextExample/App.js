import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'

import ListItems from './Components/listItems'
import AddItems from './Components/addItems'

import { PackingContext, PackingDefaults } from './Context/packingContext'

export default class App extends Component {
  constructor (props) {
    super(props)
    this.state = {
      ...PackingDefaults,
      addItem: this.addItem,
      setNewItemName: this.setNewItemName,
      clear: this.clear
    }
  }

  addItem = () => {
    this.setState((state) => ({
      allItems: [...state.allItems, state.newItemName],
      newItemName: ''
    }))
  }

  setNewItemName = (value) => {
    this.setState({newItemName: value})
  }

  clear = () => {
    this.setState({allItems: []})
  }

  render () {
    return (
      <ScrollView
        contentContainerStyle={styles.container}
        keyboardShouldPersistTaps='always'
      >
        <Text style={styles.welcome}>
          Welcome to React 16 Context
        </Text>
        <PackingContext.Provider value={this.state}>
          <AddItems />
          <ListItems />
        </PackingContext.Provider>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  }
})
