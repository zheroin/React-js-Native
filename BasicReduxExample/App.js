/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native';
import ListItems from './Components/listItems'
import AddItems from './Components/addItems'
import { Provider } from 'react-redux'
import configureStore from './Redux/Store/configureStore'
let store = configureStore()

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ScrollView
          contentContainerStyle={styles.container}
          keyboardShouldPersistTaps='always'
        >
          <Text style={styles.welcome}>
            Welcome to Redux
          </Text>
          <AddItems />
          <ListItems />
        </ScrollView>
      </Provider>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,

    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
