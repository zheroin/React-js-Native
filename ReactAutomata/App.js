import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'
import { withStatechart } from 'react-automata'
import statechart from './Statecharts/index'

import ListItems from './Components/listItems'
import AddItems from './Components/addItems'

class App extends Component {
  constructor (props) {
    super(props)
    this.state = {
      allItems: ['nachos', 'burritos', 'hot dog'],
      newItemName: ''
    }
  }

  addItem = () => {
    this.setState((state) => ({
      allItems: [...state.allItems, state.newItemName],
      newItemName: ''
    }))
  }

  setNewItemName = (value) => {
    this.setState({newItemName: this.props.value})
  }

  clear = () => {
    this.setState({allItems: []})
  }

  handAddItem = () => this.props.transition('ADD_ITEM')
  handleSetNewItemName = value => this.props.transition('SET_NEW_ITEM_NAME', {value})
  handleClear = () => this.props.transition('CLEAR')

  render () {
    return (
      <ScrollView
        contentContainerStyle={styles.container}
        keyboardShouldPersistTaps='always'
      >
        <Text style={styles.welcome}>
          Welcome to setState + react-automata
        </Text>
        <AddItems
          addItem={this.handAddItem}
          setNewItemName={this.handleSetNewItemName}
          newItemName={this.state.newItemName}
          clear={this.handleClear}
        />
        <ListItems
          allItems={this.state.allItems}
        />
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})

export default withStatechart(statechart)(App)
