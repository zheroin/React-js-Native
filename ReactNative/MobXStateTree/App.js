import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'

import { Provider } from "mobx-react"
import ListItems from './Components/listItems'
import AddItems from './Components/addItems'

import { ListStoreModel } from './MST/listStore'

// first param of create is initial state (Optionals take care)
// second param is dependency injection(s)
const listStore = ListStoreModel.create()

export default class App extends Component {
  render () {
    // Provider means anything can grab via @inject()
    // Will possibly leave in future versions of MST due to context
    return (
      <Provider
        listStore={listStore}
      >
        <ScrollView
          contentContainerStyle={styles.container}
          keyboardShouldPersistTaps='always'
        >
          <Text style={styles.welcome}>
            Welcome to MobX-State-Tree
          </Text>
          <AddItems />
          <ListItems />
        </ScrollView>
      </Provider>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})
