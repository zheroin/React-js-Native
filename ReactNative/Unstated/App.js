import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native';
import ListItems from './Components/listItems'
import { Provider } from "unstated"
import AddItems from './Components/addItem'

export default class App extends Component {
  render () {
    return (
      <Provider>
        <ScrollView
          contentContainerStyle={styles.container}
          keyboardShouldPersistTaps='always'
        >
          <Text style={styles.welcome}>
            Welcome to unstated
          </Text>
          <AddItems />
          <ListItems />
        </ScrollView>
      </Provider>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})
