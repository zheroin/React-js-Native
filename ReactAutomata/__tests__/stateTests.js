import { testStatechart } from 'react-automata'
import App from '../App'
import statechart from '../Statecharts/index'

test('all state snapshots', () => {
  testStatechart({ statechart }, App)
})
