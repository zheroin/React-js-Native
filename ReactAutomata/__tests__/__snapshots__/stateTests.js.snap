// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all state snapshots: idle 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "padding": 20,
    }
  }
  keyboardShouldPersistTaps="always"
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to setState + react-automata
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Current State 'idle'
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        clearTextOnFocus={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Packing Item"
        returnKeyLabel="Add Item"
        returnKeyType="done"
        style={
          Object {
            "backgroundColor": "#ddf",
            "flex": 1,
            "paddingHorizontal": 4,
          }
        }
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#cfc",
            "borderRadius": 4,
            "margin": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ADD
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#faa",
            "borderRadius": 4,
            "margin": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          CLEAR
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "blue",
            "padding": 1,
            "paddingHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          nachos
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "blue",
            "padding": 1,
            "paddingHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          burritos
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "blue",
            "padding": 1,
            "paddingHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          hot dog
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`all state snapshots: loaded 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "padding": 20,
    }
  }
  keyboardShouldPersistTaps="always"
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to setState + react-automata
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Current State 'idle'
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        clearTextOnFocus={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Packing Item"
        returnKeyLabel="Add Item"
        returnKeyType="done"
        style={
          Object {
            "backgroundColor": "#ddf",
            "flex": 1,
            "paddingHorizontal": 4,
          }
        }
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#cfc",
            "borderRadius": 4,
            "margin": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ADD
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#faa",
            "borderRadius": 4,
            "margin": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          CLEAR
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "blue",
            "padding": 1,
            "paddingHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          nachos
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "blue",
            "padding": 1,
            "paddingHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          burritos
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "blue",
            "padding": 1,
            "paddingHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          hot dog
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
